### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by fabxc**
@brancz

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main // import "github.com/kubernetes/kube-state-metrics"
+package main // `import "k8s.io/kube-statemetric"
```
Comment (brancz) [inline main.go]: ``` ` ``` -> `"`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+deps:
+	go get github.com/tools/godep
+
+build: clean deps
+	$(ENVVAR) godep go build -o kube-state-metrics 
+
+test-unit: clean deps build
+	$(ENVVAR) godep go test --race . $(FLAGS)
+
+container: build
```
Comment (brancz) [inline Makefile]: Maybe `container-image`? The make target `container` rather sounds like creating a running container.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,26 @@
+all: build
+
+FLAGS =
+ENVVAR = GOOS=linux GOARCH=amd64 CGO_ENABLED=0
+REGISTRY = gcr.io/google_containers
+TAG = v0.2.0
```
Comment (brancz) [inline Makefile]: Wdyt about moving this into a file? I can see this easily being forgotten when creating a new release...

File: main.go (hunk 1 above)
Comment (brancz) [inline main.go]: And typo `kube-statemetric` -> `kube-state-metrics`

File: Makefile (hunk 2 above)
Comment (fabxc) [inline Makefile]: The file was largely taken from the rescheduler repository. Figured we just keep it consistent.

File: Makefile (hunk 3 above)
Comment (fabxc) [inline Makefile]: Same as above. Let's move it into a file later if it turns out to be okay.

**Comment by fabxc** [comment]
Fixed import path.

File: Makefile (hunk 2 above)
Comment (brancz) [inline Makefile]: sgtm

**Comment by w0mbat** [comment]
I had used the `centurylink/golang-builder` container to build the binary and package everything with into a Docker image - do we want to update `Building the Docker container` section and describe it with the `make` command?

**Comment by brancz** [comment]
@w0mbat that would be good I guess /cc @fabxc

**Pull request merged**